pydantic/main.py

```diff
@@ -287,13 +287,28 @@ def __new__(mcs, name, bases, namespace, **kwargs):  # noqa C901
         else:
             json_encoder = pydantic_encoder
         pre_rv_new, post_rv_new = extract_root_validators(namespace)
+
+        dedupe_pre_rv = set([])
```

Comment:
why is this required? Maybe it's a separate issue.

Reply:
I explained this in my original "essay" - "It turns out deduplication of root validators is necessary in the metaclass, since it appears __new__ is called twice for generics, once at class definition, and another when type arguments are applied (seems like semantically python is creating an on-the-fly subclass with the type variables unified,)"

The rest of the paragraph goes into further detail describing that this is a current bug in pydantic which is hidden by the current behavior of stopping on the first root validator to fail. In short, you are currently duplicating root validators on generic models.

I gotta say, I can appreciate you maintain this project in addition to your whole real life, and that you do it probably for free, but please consider that my contribution here is under _exactly_ those same circumstances. Your last comment comes off hostile, I'm not really sure why that's warranted.